providers/openmetrics/client_metrics.go

```diff
@@ -1,99 +1,103 @@
 package metrics
 
 import (
-	"github.com/prometheus/client_golang/prometheus"
 	openmetrics "github.com/prometheus/client_golang/prometheus"
 )
 
 // ClientMetrics represents a collection of metrics to be registered on a
 // Prometheus metrics registry for a gRPC client.
 type ClientMetrics struct {
-	clientRegister openmetrics.Registerer
-
 	clientStartedCounter    *openmetrics.CounterVec
 	clientHandledCounter    *openmetrics.CounterVec
 	clientStreamMsgReceived *openmetrics.CounterVec
 	clientStreamMsgSent     *openmetrics.CounterVec
 
-	clientHandledHistogramEnabled bool
-	clientHandledHistogramOpts    openmetrics.HistogramOpts
-	clientHandledHistogram        *openmetrics.HistogramVec
-
-	clientStreamRecvHistogramEnabled bool
-	clientStreamRecvHistogramOpts    openmetrics.HistogramOpts
-	clientStreamRecvHistogram        *openmetrics.HistogramVec
-
-	clientStreamSendHistogramEnabled bool
-	clientStreamSendHistogramOpts    openmetrics.HistogramOpts
-	clientStreamSendHistogram        *openmetrics.HistogramVec
+	// clientHandledHistogram can be nil
+	clientHandledHistogram *openmetrics.HistogramVec
+	// clientStreamRecvHistogram can be nil
+	clientStreamRecvHistogram *openmetrics.HistogramVec
+	// clientStreamSendHistogram can be nil
+	clientStreamSendHistogram *openmetrics.HistogramVec
```

Comment:
Use `nil` a the flag rather than a separate field. Better memory alignment and smaller object size.